### ISSUE TITLE
interpretation of speed is a bit unclear

### ISSUE THREAD

**Issue by David-Levinthal** [facebookresearch/fairseq-py#70]
this is an information request  :-)
the output from running a seq2seq model
 using 1 GPUs (with max tokens per GPU = 4000 and max sentences per GPU = None)
| model lstm_luong_wmt_en_de, criterion CrossEntropyCriterion
| epoch 001:   1000 / 37483 loss=9.68 (10.49), wps=4632, wpb=3514, bsz=126, lr=0.25, clip=100%, gnorm=1.02492
| epoch 001:   2000 / 37483 loss=9.72 (9.95), wps=4616, wpb=3512, bsz=123, lr=0.25, clip=100%, gnorm=0.876077
| epoch 001:   3000 / 37483 loss=7.90 (9.48), wps=4627, wpb=3513, bsz=122, lr=0.25, clip=100%, gnorm=0.841918
| epoch 001:   4000 / 37483 loss=8.97 (9.05), wps=4630, wpb=3520, bsz=122, lr=0.25, clip=100%, gnorm=0.820904

prints a words/sec but is this input words, output words or input+ output?
It would be useful if one could get sentences/sec as well.
this is particularly difficult as the batch size is set in tokens.
making "standardization and performance comparisons difficult 
batch size for TF/NMT, MXNET/Sockeye are in sentences.
is there a sentence bucketing done?, How does one get tokens/training sentence, assuming short sentences are not used. Is there a cutoff on sentence length?
thanks
d

**Comment by David-Levinthal** [comment]
I should perhaps also note that an LSTM cell executes 8*2*hidden_size*hidden_size FP ops in the forward pass if the hidden size is large. see: https://github.com/David-Levinthal/machine-learning/blob/master/Introduction%20to%20machine%20learning%20algorithms.pdf for measurements showing this.
a factor of 3 can be applied to include backpropagation
by knowing the input and output WPS rates and the encoder and decoder layer count (and if the encoder has a bidirectional first layer or maybe more)
one can compute the flop rate achieved by the HW.

**Comment by David-Levinthal** [comment]
8*2*hidden_size*hidden_size   keyboard issues

**Comment by David-Levinthal** [comment]
8 X 2 X hidden_size*hidden_size github doesn't like stars between digits

**Comment by myleott** [comment]
You can either generate batches based on the number of tokens with `--max-tokens` or batch by number of sentences with `--max-sentences`. Typically batching by number of tokens is more memory efficient.

While we don't output sentences per second directly, you can compute it based on the output fields `wps`, `wpb` and `bsz`:
```
batches_per_second = wps / wpb
sentences_per_second = bsz * batches_per_second
```

**Comment by David-Levinthal** [comment]
Is wps input words only?

**Comment by myleott** [comment]
Target words only.

**Comment by David-Levinthal** [comment]
Would it be easy to print both source and target wps ? That would make flop
rates easy to calculate 😁

**Comment by michaelauli** [comment]
Please take a look in the code how the output for wps etc. is built to figure out how to add this information.

**Comment by David-Levinthal** [comment]
will do..
